tests/integration/test_dq_rules.py

```diff
@@ -0,0 +1,153 @@
+        gcp_project_id,
+        target_bq_result_dataset_name,
+        target_bq_result_table_name,):
+
+        client = client.get_connection()
+
+        table_id = f"{gcp_project_id}.{target_bq_result_dataset_name}.{target_bq_result_table_name}_expected"
+
+        job_config = bigquery.LoadJobConfig(
+            source_format=bigquery.SourceFormat.CSV,
+            skip_leading_rows=1,
+            autodetect=True,
+            write_disposition="WRITE_TRUNCATE")
+
+        with open(test_resources / "expected_results.csv", "rb") as source_file:
+            job = client.load_table_from_file(source_file, table_id, job_config=job_config)
+
+        job.result()  # Waits for the job to complete.
+
+        table = client.get_table(table_id)  # Make an API request.
+        print(f"Loaded {table.num_rows} rows and {len(table.schema)} columns to {table_id}")
+
+    def test_dq_rules(
+        self,
+        runner,
+        temp_configs_dir,
+        gcp_application_credentials,
+        gcp_project_id,
+        gcp_bq_dataset,
+        gcp_bq_region,
+        target_bq_result_dataset_name,
+        target_bq_result_table_name,
+        tmp_path,
+        client,
+        gcp_impersonation_credentials,
+        gcp_sa_key,
+        create_expected_results_table,
+    ):
+
+        try:
+            temp_dir = Path(tmp_path).joinpath("clouddq_test_cli_integration_4")
+            temp_dir.mkdir(parents=True)
+            with working_directory(temp_dir):
+                logger.info(f"test_last_modified_in_dq_summary {gcp_application_credentials}")
+                target_table = f"{gcp_project_id}.{target_bq_result_dataset_name}.{target_bq_result_table_name}"
+                args = [
+                    "T1_DQ_1_VALUE_NOT_NULL,T2_DQ_1_EMAIL,T3_DQ_1_EMAIL_DUPLICATE",
+                    f"{temp_configs_dir}",
+                    f"--gcp_project_id={gcp_project_id}",
+                    f"--gcp_bq_dataset_id={gcp_bq_dataset}",
+                    f"--gcp_region_id={gcp_bq_region}",
+                    f"--target_bigquery_summary_table={target_table}",
+                    "--debug",
+                    "--summary_to_stdout",
+                    ]
+                result = runner.invoke(main, args)
+                print(result.output)
+                assert result.exit_code == 0
+
+                # Prepare dbt runtime
+                dbt_runner = DbtRunner(
+                    dbt_path=None,
+                    dbt_profiles_dir=None,
+                    environment_target="Dev",
+                    gcp_project_id=gcp_project_id,
+                    gcp_region_id=gcp_bq_region,
+                    gcp_bq_dataset_id=gcp_bq_dataset,
+                    gcp_service_account_key_path=gcp_sa_key,
+                    gcp_impersonation_credentials=gcp_impersonation_credentials,
+                )
+                dbt_path = dbt_runner.get_dbt_path()
+                invocation_id = get_dbt_invocation_id(dbt_path)
```

Comment:
interesting that you had to do this to get the invocation id -- are there any risk that you might retrieve the wrong invocation id here?

Reply:
I can't think of any risk for now, as it will get the invocation Id as the dbt run completes successfully. We can discuss in details.